packages/neuron-ui/src/components/WalletEditor/index.tsx

```diff
@@ -1,65 +1,42 @@
+import React, { useEffect, useMemo } from 'react'
 import { RouteComponentProps } from 'react-router-dom'
-import { Card, Form, Button } from 'react-bootstrap'
+import { Card, Form, Button, Col, Row } from 'react-bootstrap'
 import { useTranslation } from 'react-i18next'
 
 import { ContentProps } from '../../containers/MainContent'
 import InlineInput, { InputProps } from '../../widgets/InlineInput'
-import { MainActions } from '../../containers/MainContent/reducer'
+import { MainActions, actionCreators } from '../../containers/MainContent/reducer'
 import { useNeuronWallet } from '../../utils/hooks'
-import InputWalletPasswordDialog, { CheckType } from '../Settings/InputWalletPasswordDialog'
 import Dialog from '../../widgets/Dialog'
-import { Routes } from '../../utils/const'
+
+import { useWalletEditor, useInputs, useIsParamsValid, useToggleDialog } from './hooks'
 
 export default (props: React.PropsWithoutRef<ContentProps & RouteComponentProps<{ id: string }>>) => {
   const { match, dialog, dispatch } = props
-  const { params } = match
+  const {
+    params: { id },
+  } = match
   const [t] = useTranslation()
   const {
     settings: { wallets },
-    messages: errorMessages,
   } = useNeuronWallet()
 
-  const myWallet = wallets.find(wallet => wallet.id === params.id)
-  const [walletName, setWalletName] = useState('')
-  const [password, setPassword] = useState('')
-  const [confirmPassword, setConfirmPassword] = useState('')
+  const wallet = useMemo(() => wallets.find(w => w.id === id), [id])
+
+  if (!wallet) {
+    // TODO: Better error handling
+    throw new Error('Wallet not found')
+  }
+
+  const editor = useWalletEditor()
 
   useEffect(() => {
-    if (password !== '') {
-      dispatch({
-        type: MainActions.SetDialog,
-        payload: {
-          open: false,
-        },
-      })
-      props.history.push(`${Routes.SettingsWallets}`)
-    }
-  }, [wallets])
+    editor.initiate(wallet.name)
+  }, [id])
 
-  const inputs: InputProps[] = [
-    {
-      label: t('settings.wallet-manager.edit-wallet.wallet-name'),
-      value: walletName,
-      onChange: e => setWalletName(e.currentTarget.value),
-      placeholder: myWallet!.name,
-      maxLength: 20,
-    },
-    {
-      label: t('settings.wallet-manager.edit-wallet.password'),
-      value: password,
-      onChange: e => setPassword(e.currentTarget.value),
-      placeholder: t('settings.wallet-manager.edit-wallet.password'),
-      inputType: 'password',
-    },
-    {
-      label: t('settings.wallet-manager.edit-wallet.confirm-password'),
-      value: confirmPassword,
-      onChange: e => setConfirmPassword(e.currentTarget.value),
-      placeholder: t('settings.wallet-manager.edit-wallet.confirm-password'),
-      inputType: 'password',
-    },
-  ]
+  const inputs: InputProps[] = useInputs(editor)
+  const isParamsValid = useIsParamsValid(editor.name.value, editor.newPassword.value, editor.confirmNewPassword.value)
```

Comment:
```suggestion
  const areParamsValid = useAreParamsValid(editor.name.value, editor.newPassword.value, editor.confirmNewPassword.value)
```

Kind of difficult to name this 😢